src/OpenSpaceToolkit/Astrodynamics/Trajectory/State.cpp

```diff
@@ -499,6 +499,31 @@ State State::inFrame(const Shared<const Frame>& aFrameSPtr) const
     };
 }
 
+bool State::isNear(const State& aState, const std::unordered_map<CoordinateSubset, Real>& aToleranceMap) const
+{
+    if (aToleranceMap.size() != coordinatesBrokerSPtr_->accessSubsets()->getSize())
+    {
+        throw ostk::core::error::runtime::Wrong("Tolerance Map");
+    }
+
+    const State deltaState = aState - *this;
+
+    bool isNear = true;
+    for (const auto& [subset, tolerance] : aToleranceMap) {
+        if (!coordinatesBrokerSPtr_->hasSubset(subset)) {
```

Comment:
Is this necessary? Won't the subtract method raise an error if the coordinate subsets are not identical?